catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx

```diff
@@ -0,0 +1,628 @@
+import React from 'react';
+import { useEffect, useReducer, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import {
+  ActionList,
+  ActionListItem,
+  Button,
+  EmptyState,
+  EmptyStateIcon,
+  Form,
+  FormGroup,
+  FormHelperText,
+  PageSection,
+  PageSectionVariants,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextArea,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import {
+  ExclamationCircleIcon,
+} from '@patternfly/react-icons';
+
+import {
+  checkSalesforceId,
+  createWorkshop,
+  createWorkshopProvision,
+} from '@app/api';
+import {
+  selectCatalogNamespace,
+  selectUserGroups,
+  selectUserIsAdmin,
+  selectUserRoles,
+  selectWorkshopNamespaces,
+} from '@app/store';
+import {
+  CatalogItem,
+  CatalogItemSpecParameter,
+  CatalogNamespace,
+  ServiceNamespace,
+  Workshop,
+  WorkshopProvision,
+} from '@app/types';
+import {
+  ConditionValues,
+  checkAccessControl,
+  checkCondition,
+  displayName,
+  randomString,
+} from '@app/util';
+
+import DynamicFormInput from '@app/components/DynamicFormInput';
+import LoadingIcon from '@app/components/LoadingIcon';
+import PatientNumberInput from '@app/components/PatientNumberInput';
+
+import './catalog-item-workshop-form.css';
+
+interface FormState {
+  conditionChecks: {
+    canceled: boolean;
+    complete: boolean;
+    running: boolean;
+  };
+  formGroups: FormStateParameterGroup[];
+  initComplete: boolean;
+  parameters: {[name: string]: FormStateParameter};
+}
+
+interface FormStateAction {
+  type: "checkConditionsComplete" | "init" | "parameterUpdate";
+  catalogItem?: CatalogItem;
+  parameterIsValid?: boolean;
+  parameterName?: string;
+  parameterValue?: boolean|number|string|undefined;
+}
+
+interface FormStateParameter {
+  default?: boolean|number|string|undefined;
+  isDisabled?: boolean;
+  isHidden?: boolean;
+  isRequired?: boolean;
+  // isValid is specifically the result of component validation such as min/max on numeric input
+  isValid?: boolean;
+  name: string;
+  spec: CatalogItemSpecParameter;
+  // validationMessage and validationResult are set by checking validation condition
+  validationMessage?: string|undefined;
+  validationResult?: boolean|undefined;
+  value?: boolean|number|string|undefined;
+}
+
+interface FormStateParameterGroup {
+  formGroupLabel: string;
+  isRequired?: boolean;
+  key: string;
+  parameters: FormStateParameter[];
+}
+
+function cancelFormStateConditionChecks(state:FormState): void {
+  if (state) {
+    state.conditionChecks.canceled = true;
+  }
+}
+
+// Because salesforce checks are asynchronous they need to be resolved before checking the condition logic
+async function _checkCondition(condition: string, vars: ConditionValues): Promise<boolean> {
```

Comment:
checkSalesforceCondition could be a better naming?

Reply:
It is checking more than just the salesforce value. It just just that the saleforce check requires an asynchronous call.